velox/vector/VectorPool.h

```diff
@@ -62,4 +63,26 @@ class VectorPool {
   std::array<TypePool, kNumCachedVectorTypes> vectors_;
 };
 
+/// A simple vector ptr wrapper with an associated context object. It releases
+/// the allocated vector back to vector pool in context on destruction.
+///
+/// NOTE: the context object must provide 'releaseVector' vector such as
+/// EvalCtx.
+template <class Context>
+class ScopedVectorPtr {
+ public:
+  explicit ScopedVectorPtr(Context& context) : context_(context) {}
```

Comment:
This is tricky. what does it mean to move this, and what does it mean to copy this? 
Also would be nice to include Recycle in the naming. 
The reason for adding this is to avoid tracking all function exit locations if i  understand/
I wonder if a better approach would be 

VectorPtr ptr;
VectorRecycler(ptr, context);
or 
RecyleAtExit(ptr, context);

in this way, the code changes are minimal no need to access ptr() from within this. 
in fact, this guy is not really introduced to own or hold the ptr. but just to release it at the exit.

```
class RecycleAtExit {
RecycleAtExit (const RecycleAtExit&) = delete;
RecycleAtExit& operator= (const RecycleAtExit&) = delete;

RecycleAtExit (const RecycleAtExit&&) = delete;
RecycleAtExit& operator= (const RecycleAtExit&&) = delete;

// also delete assignment operators explicitly. 

 ~ScopedVectorPtr() {
    context_.releaseVector(vector_);
  }
 private:
  VectorPtr& vector_;
  Context& context_;
};
```

Reply:
This approach looks nice.